src/controllers/otp.js

```diff
@@ -0,0 +1,43 @@
+const otpGenerator = require("otp-generator");
+const OTP = require("../models/Otp");
+const User = require("../models/User");
+const asyncHandler = require("express-async-handler");
+
+const sendOTP = asyncHandler(async (req, res) => {
+  const { email } = req.body;
+
+  const isRegistered = await User.findOne({ email });
+
+  if (isRegistered)
+    return res.status(401).json({
+      success: false,
+      message: "User is already registered",
+    });
+
+  let otp = otpGenerator.generate(4, {
+    upperCaseAlphabets: false,
+    lowerCaseAlphabets: false,
+    specialChars: false,
+  });
+
+  let result = await OTP.findOne({ otp });
+
+  while (result) {
```

Comment:
Hey man, really nice work honestly. I am confident we can get good work done with this. 

I have a concern here though. Why are we checking if the OTP has been generated before? Would it matter if two users had the same OTP? 
If that wouldn't matter, then I think we should without the db call to find one OTP and the while condition to continuously create OTPs until it is unique

Reply:
True, I just changed the logic for the otp to regenerated only if it has sent the same otp to the same user before